### PR TITLE
Cache access group members as secrets

### PR DESCRIPTION
**Pull request by lgruen**
To work around the incredibly slow Google Groups API.

### CODE REVIEW

File: stack/__main__.py [hunk 1]
```diff
@@ -315,6 +358,7 @@ def add_bucket_permissions(
         member=pulumi.Output.concat('serviceAccount:', WEB_SERVER_SERVICE_ACCOUNT),
     )
 
+    # TODO(@lgruen): remove this once secrets are used for checking memberships.
```
Comment (illusional) [inline stack/__main__.py:361]: Should these be removed now?

File: stack/__main__.py [hunk 2]
```diff
@@ -325,6 +369,7 @@ def add_bucket_permissions(
         roles=[gcp.cloudidentity.GroupMembershipRoleArgs(name='MEMBER')],
     )
 
+    # TODO(@lgruen): remove this once secrets are used for checking memberships.
```
Comment (illusional) [inline stack/__main__.py:372]: This one too

**Review by illusional — COMMENTED** [review]
2 things I don't quite understand:

1. Is this just an initial test, because I don't see a web server implementation, and there are a few TODOs.
2. I can see `--trigger-topic=access-group-cache-refresh` when deploying the function, is that topic automatically triggered on an access-group change, or are there plans to invalidate it like once a day or something (you suggest a cron job in the README).

**Comment by lgruen** [comment]
> 1. Is this just an initial test, because I don't see a web server implementation, and there are a few TODOs.

Not a test, this is just a first PR in a series of changes. I didn't want to send you a few thousand lines of code at once and these can be nicely staged.

> 2. I can see `--trigger-topic=access-group-cache-refresh` when deploying the function, is that topic automatically triggered on an access-group change, or are there plans to invalidate it like once a day or something (you suggest a cron job in the README).

Cloud Scheduler sends a message to that topic every 5 minutes, which triggers the execution of this function (that's basically the Cron job).

File: stack/__main__.py (hunk 1 above)
Comment (lgruen) [reply stack/__main__.py:361]: Not yet, we need to update the analysis-runner / web server first.

File: stack/__main__.py (hunk 2 above)
Comment (lgruen) [reply stack/__main__.py:372]: Not yet, we need to update the analysis-runner / web server first.

**Review by illusional — APPROVED** [review]

**Pull request merged**